components/public-api-server/pkg/oidc/oauth2.go

```diff
@@ -73,17 +74,21 @@ func (s *Service) OAuth2Middleware(next http.Handler) http.Handler {
 			http.Error(rw, "bad state param", http.StatusBadRequest)
 			return
 		}
+		useHttpErrors := state.UseHttpErrors
```

Comment:
Can we not do this based on the ContentType which is being requested on the `/callback` endpoint?

Reply:
An in general, what prevents us from always responding with the appropriate HTTP status code, but use the ContentType to determine whether to respond with JSON, or text?

Reply:
Simple answer: the callback route isn't called by us, thus we cannot rely on HTTP headers to decide on that. 

Other way around, the fact that we're not interested in HTTP status code makes me think if we could drop that here.